### ISSUE TITLE
crf_losses.py as in keras_contrib?

### ISSUE THREAD

**Issue by ahmad-alismail** [tensorflow/addons#2363]
Hello, I want to ask if you there is an alternative crf_losses.py as in keras_contrib in tensorflow_addons?

**Comment by Harsh188** [comment]
Hi @afi1289,

I believe that we have a [crf layer](https://github.com/tensorflow/addons/blob/master/tensorflow_addons/layers/crf.py) however we do not have an alternative for crf losses.

**Comment by afi1289** [comment]
Thank you for your reply @Harsh188 
Could you add crf_losses.py to losses, crf_accuracies.py to metrics? thats will help a lot (especially for beginners like me) by implementing BiLSTM-CRF models which initially implemented using keras-contrib. 
Thank you in advance!

**Comment by kankajan814** [comment]
Hi @Harsh188 ,
I agree with @afi1289 , please add them

**Comment by bhack** [comment]
You can check https://github.com/xuxingya/tf2crf

I don't know if @xuxingya Is interested to contribute here.

**Comment by jonpsy** [comment]
@bhack I'm ready!

**Comment by jonpsy** [comment]
Okay, I did some research on this. Thing is, the loss function doesn't solely depend on y_true and y_pred but also the internal variables. So, we **can't use eager tensor** for y_pred, more at this issue: https://github.com/tensorflow/tensorflow/issues/37119

I could go that path, depending on whether the community agrees on the repercussions.

**Comment by ashutosh1919** [comment]
@afi1289 , can this be closed as part of #1999 ?

**Comment by howl-anderson** [comment]
> If you won't mind, I'd like to take a stab at this. I've found a [repo](https://github.com/howl-anderson/tf_crf_layer) which implements CRF Loss.
> 
> CRF Loss: https://github.com/howl-anderson/tf_crf_layer/blob/master/tf_crf_layer/loss/crf_loss.py
> CRF Accuracy: https://github.com/howl-anderson/tf_crf_layer/blob/master/tf_crf_layer/metrics/crf_accuracy.py
> 
> A bit of polish, and should make for a good PR, I suppose?

I am the author of this repo, and I happy to bring back the CRF accuracy feature to TFA. I will submit a PR for it when it is ready. 

`constraints` in the AllenNLP (https://docs.allennlp.org/main/api/modules/conditional_random_field/) is awesome feature too. It also exits in my repo (in a different branch). I hope I can bring it to the TFA in the future.

**Comment by seanpmorgan** [comment]
TensorFlow Addons is transitioning to a minimal maintenance and release mode. New features will not be added to this repository. For more information, please see our public messaging on this decision:
[TensorFlow Addons Wind Down](https://github.com/tensorflow/addons/issues/2807)

Please consider sending feature requests / contributions to other repositories in the TF community with a similar charters to TFA:
[Keras](https://github.com/keras-team/keras)
[Keras-CV](https://github.com/keras-team/keras-cv)
[Keras-NLP](https://github.com/keras-team/keras-nlp)
